rclcpp_components/src/component_container_isolated.cpp

```diff
@@ -13,37 +13,57 @@
 // limitations under the License.
 
 #include <memory>
-#include <vector>
 #include <string>
+#include <vector>
 
 #include "rclcpp/rclcpp.hpp"
 #include "rclcpp/utilities.hpp"
 #include "rclcpp_components/component_manager_isolated.hpp"
 
 int main(int argc, char * argv[])
 {
-  /// Component container with dedicated single-threaded executors for each components.
+  /// Component container with dedicated executor for each component
   rclcpp::init(argc, argv);
+
   // parse arguments
-  bool use_multi_threaded_executor{false};
+  // valid entries: --executor-type single-threaded, --executor-type multi-threaded, --executor-type events
   std::vector<std::string> args = rclcpp::remove_ros_arguments(argc, argv);
-  for (auto & arg : args) {
-    if (arg == std::string("--use_multi_threaded_executor")) {
-      use_multi_threaded_executor = true;
+
+  std::string executor_type = "single-threaded";  // default
+  for (size_t i = 0; i < args.size(); ++i) {
+    if (args[i] == "--executor-type") {
+      if (i + 1 < args.size()) {
+        executor_type = args[i + 1];
+        break;
+      }
+    } else if (args[i] == "--use_multi_threaded_executor") { // backward compatibility
+      executor_type = "multi-threaded";
     }
   }
+
   // create executor and component manager
-  auto exec = std::make_shared<rclcpp::executors::SingleThreadedExecutor>();
   rclcpp::Node::SharedPtr node;
-  if (use_multi_threaded_executor) {
-    using ComponentManagerIsolated =
-      rclcpp_components::ComponentManagerIsolated<rclcpp::executors::MultiThreadedExecutor>;
+  std::shared_ptr<rclcpp::Executor> exec;
+  if (executor_type == "events") {
+    using executor = rclcpp::experimental::executors::EventsExecutor;
+    using ComponentManagerIsolated = rclcpp_components::ComponentManagerIsolated<executor>;
+    exec = std::make_shared<executor>();
     node = std::make_shared<ComponentManagerIsolated>(exec);
-  } else {
-    using ComponentManagerIsolated =
-      rclcpp_components::ComponentManagerIsolated<rclcpp::executors::SingleThreadedExecutor>;
+  } else if (executor_type == "multi-threaded") {
+    using executor = rclcpp::executors::MultiThreadedExecutor;
+    using ComponentManagerIsolated = rclcpp_components::ComponentManagerIsolated<executor>;
+    exec = std::make_shared<executor>();
+    node = std::make_shared<ComponentManagerIsolated>(exec);
+  } else if (executor_type == "single-threaded") {
+    using executor = rclcpp::executors::SingleThreadedExecutor;
+    using ComponentManagerIsolated = rclcpp_components::ComponentManagerIsolated<executor>;
+    exec = std::make_shared<executor>();
     node = std::make_shared<ComponentManagerIsolated>(exec);
+  } else {
+    std::cerr << "Invalid executor type: " << executor_type << std::endl;
```

Comment:
this is actually behavior change compared to before. before, even if user specify the wrong option, that falls back to `SingleThreadedExecutor`, but now it fails and user does not really know what can be set. it would be probably nice to show what and how executor types are set? or command line helper function can be introduced?